src/torchmetrics/functional/segmentation/mean_iou.py

```diff
@@ -96,17 +111,27 @@ def mean_iou(
         The mean IoU score
 
     Example:
+        >>> import torch
         >>> from torch import randint
         >>> from torchmetrics.functional.segmentation import mean_iou
-        >>> preds = randint(0, 2, (4, 5, 16, 16))  # 4 samples, 5 classes, 16x16 prediction
-        >>> target = randint(0, 2, (4, 5, 16, 16))  # 4 samples, 5 classes, 16x16 target
-        >>> mean_iou(preds, target, num_classes=5)
-        tensor([0.3193, 0.3305, 0.3382, 0.3246])
-        >>> mean_iou(preds, target, num_classes=5, per_class=True)
-        tensor([[0.3093, 0.3500, 0.3081, 0.3389, 0.2903],
-                [0.2963, 0.3316, 0.3505, 0.2804, 0.3936],
-                [0.3724, 0.3249, 0.3660, 0.3184, 0.3093],
-                [0.3085, 0.3267, 0.3155, 0.3575, 0.3147]])
+        >>> preds = randint(0, 2, (4, 5, 16, 16), generator=torch.Generator().manual_seed(42))
+        >>> target = randint(0, 2, (4, 5, 16, 16), generator=torch.Generator().manual_seed(43))
```

Comment:
```suggestion
        >>> preds = randint(0, 2, (4, 5, 16, 16), generator=torch.Generator().manual_seed(42))  # 4 samples, 5 classes, 16x16 prediction
        >>> target = randint(0, 2, (4, 5, 16, 16), generator=torch.Generator().manual_seed(43))  # 4 samples, 5 classes, 16x16 target
```
i would prefer if we re-added the comments here